### PR TITLE
Add possibility to override feature flags with environment variable FEATURE_FLAGS

### PR DESCRIPTION
**Pull request by ofreyssinet-ledger**
### 📝 Description

I did this because I needed to set a feature flag in a playwright test and doing it through an env variable made sense as playwright fixtures are already modifying `env`.

Now in playwright you can do this to enabled a feature flag:
```ts
test.use({ featureFlags: { mockFeature: { enabled: true } } });
```

**It is working both on LLM and LLD**.

I also made the appropriate changes in the feature flag debugging UIs in both LLM and LLD.

I'm sure this will be useful in the future for other purposes.

### ❓ Context

- **Impacted projects**: `ledger-live-mobile` `ledger-live-desktop` `types-live` `live-common` 
- **Linked resource(s)**: [LIVE-3895] 

### ✅ Checklist

- [ ] **Test coverage** 
- [x] **Atomic delivery** 
- [x] **No breaking changes** 

#### Compiling LLM with `.env` file in `apps/ledger-live-mobile`:
Screenshot 2022-09-26 at 16 29 21

#### Running LLD like this

```sh
FEATURE_FLAGS='{"mockFeature": {"enabled": false, "someRandomKey": "aCustomValue"}}' pnpm dev:lld
```
Screenshot 2022-09-26 at 16 51 24

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

[LIVE-3895]: https://ledgerhq.atlassian.net/browse/LIVE-3895?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/ledger-live-desktop/src/renderer/components/FirebaseFeatureFlags.tsx [hunk 1]
```diff
@@ -26,6 +27,17 @@ export const FirebaseFeatureFlagsProvider = ({ children }: Props): JSX.Element =
           return localOverrides[key];
         }
 
+        const envFlags = getEnv("FEATURE_FLAGS") as { [key in FeatureId]?: Feature } | undefined;
+        if (allowOverride && envFlags) {
+          const feature = envFlags[key];
+          if (feature)
+            return {
+              ...feature,
+              overridesRemote: true,
```
Comment (gre) [inline apps/ledger-live-desktop/src/renderer/components/FirebaseFeatureFlags.tsx:36]: if it's overriden by env, should remove be false? 🤔

File: apps/ledger-live-desktop/src/renderer/screens/settings/sections/Developer/FeatureFlagsButton.tsx [hunk 2]
```diff
@@ -179,63 +184,68 @@ const FeatureFlagsButton = () => {
...
+                          <Tag active mx={1} type="outlinedOpacity" size="small">
+                            disabled for current language
+                          </Tag>
+                        ) : null}
+                      </Flex>
+                    </OldButton>
+                    {focusedName === flagName ? (
+                      <Flex flexDirection="column" pl={6} rowGap={3}>
+                        <EditSection
+                          value={inputValues[flagName] || JSON.stringify(pureValue)}
```
Comment (gre) [inline apps/ledger-live-desktop/src/renderer/screens/settings/sections/Developer/FeatureFlagsButton.tsx:226]: prefer to cache this with a memo. `stringify` every time it re-renders isn't ideal. you may want to have an intermediary component to ease this as it's included in a "if", generally speaking the JSX tree is quite deep here and splitting into more components isn't bad.

**Review by gre — COMMENTED** [review]
some feedback. good overall.

File: apps/ledger-live-desktop/src/renderer/screens/settings/sections/Developer/FeatureFlagsButton.tsx (hunk 2 above)
Comment (ofreyssinet-ledger) [reply apps/ledger-live-desktop/src/renderer/screens/settings/sections/Developer/FeatureFlagsButton.tsx:226]: Thanks I'll do that 👍

File: apps/ledger-live-desktop/src/renderer/components/FirebaseFeatureFlags.tsx (hunk 1 above)
Comment (ofreyssinet-ledger) [reply apps/ledger-live-desktop/src/renderer/components/FirebaseFeatureFlags.tsx:36]: It still overrides the remote value, if the name of the variable was different I would have done what you are suggest

File: apps/ledger-live-desktop/src/renderer/screens/settings/sections/Developer/FeatureFlagsButton.tsx (hunk 2 above)
Comment (ofreyssinet-ledger) [reply apps/ledger-live-desktop/src/renderer/screens/settings/sections/Developer/FeatureFlagsButton.tsx]: Done 👌
I split it in 3 components on both LLM & LLD -> `FeatureFlagsSettings/index.tsx` `FeatureFlagDetails.tsx` `FeatureFlagEdit.tsx`.
It's much clearer now I think

**Review by thomasrogerlux — APPROVED** [review]

**Comment by ofreyssinet-ledger** [comment]
Force push: just a squash & remerge, then did some fixes on both apps

**Review by alexandremgo — APPROVED** [review]

**Review by nabil-brn — APPROVED** [review]

**Pull request merged**